android/modules/media/src/java/android/widget/TiVideoView8.java

```diff
@@ -179,6 +179,30 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
 		}
 	}
 
+	private void constantDeprecationWarning(int constant)
+	{
+		StringBuilder warningMessage = new StringBuilder();
+		warningMessage.append("has been deprecated. Use ");
+		switch (constant) {
+			case MediaModule.VIDEO_SCALING_ASPECT_FILL:
+				warningMessage.insert(0, "Ti.Media.VIDEO_SCALING_ASPECT_FILL ");
+				warningMessage.insert(warningMessage.length(), "Ti.Media.VIDEO_SCALING_RESIZE_ASPECT_FILL");
+				break;
+			case MediaModule.VIDEO_SCALING_ASPECT_FIT:
+				warningMessage.insert(0, "Ti.Media.VIDEO_SCALING_ASPECT_FIT ");
+				warningMessage.insert(warningMessage.length(), "Ti.Media.VIDEO_SCALING_RESIZE_ASPECT_FIT");
+				break;
+			case MediaModule.VIDEO_SCALING_MODE_FILL:
+				warningMessage.insert(0, "Ti.Media.VIDEO_SCALING_MODE_FILL ");
+				warningMessage.insert(warningMessage.length(), "Ti.Media.VIDEO_SCALING_RESIZE");
+				break;
+			default:
+				return;
+		}
+		warningMessage.append(" instead.");
+		Log.w("VideoPlayerProxy", warningMessage.toString());
+	}
+
```

Comment:
The `constantDeprecationWarning()` method will end up appending strings every time the video view gets resized or a request-layout occurs. Even if you are using a non-deprecated constant. I'd prefer to avoid this. How about rewriting it to something like the below?

```java
String message = null;
final String MESSAGE_FORMAT = "%s has been deprecated. Use %s instead.";
switch (constant) {
	case MediaModule.VIDEO_SCALING_ASPECT_FILL:
		message = String.format(MESSAGE_FORMAT, "Ti.Media.VIDEO_SCALING_ASPECT_FILL", "Ti.Media.VIDEO_SCALING_RESIZE_ASPECT_FILL");
		break;
	case MediaModule.VIDEO_SCALING_ASPECT_FIT:
		message = String.format(MESSAGE_FORMAT, "Ti.Media.VIDEO_SCALING_ASPECT_FIT", "Ti.Media.VIDEO_SCALING_RESIZE_ASPECT_FIT");
		break;
	case MediaModule.VIDEO_SCALING_MODE_FILL:
		message = String.format(MESSAGE_FORMAT, "Ti.Media.VIDEO_SCALING_MODE_FILL", "Ti.Media.VIDEO_SCALING_RESIZE");
		break;
}
if (message != null) {
	Log.w("VideoPlayerProxy", message);
}
```

Reply:
That's way better, yes. I also changed the default value of `mScalingType` to use the new constant counter part.

Reply:
> I also changed the default value of mScalingType

I didn't even notice that. It's looking good now. :)